File: LabReort/Lab_5.py
# -----------------------------------------
# Program: Reverse a Given Number
# -----------------------------------------

def reverse_number(num):
    reverse = 0
    is_negative = False

    # Check if number is negative
    if num < 0:
        is_negative = True
        num = abs(num)

    # Reverse the number
    while num > 0:
        digit = num % 10
        reverse = reverse * 10 + digit
        num = num // 10

    # Restore negative sign if needed
    if is_negative:
        reverse = -reverse

    return reverse


# Main Program
num = int(input("Enter a number: "))
result = reverse_number(num)

print("Reversed number:", result)

File: LabReort/Lab_5.pl
% ----------------------------------------
% Program: Reverse a Given Number
% ----------------------------------------

% Main reverse predicate
reverse_number(Number, Reverse) :-
    Number >= 0,
    reverse_helper(Number, 0, Reverse).

% Helper predicate (Recursion)
reverse_helper(0, Reverse, Reverse).

reverse_helper(Number, Acc, Reverse) :-
    Number > 0,
    Digit is Number mod 10,
    NewAcc is Acc * 10 + Digit,
    NewNumber is Number // 10,
    reverse_helper(NewNumber, NewAcc, Reverse).

% Start program with user input
start :-
    write('Enter a number: '),
    read(Number),
    reverse_number(Number, Reverse),
    write('Reversed number is: '),
    write(Reverse), nl.
